quaddtype/numpy_quaddtype/src/casts.cpp

```diff
@@ -315,10 +501,17 @@ quad_to_numpy_strided_loop(PyArrayMethod_Context *context, char *const data[],
     char *in_ptr = data[0];
     char *out_ptr = data[1];
 
+    QuadPrecDTypeObject *quad_descr = (QuadPrecDTypeObject *)context->descriptors[0];
+    QuadBackendType backend = quad_descr->backend;
+
+    size_t elem_size = (backend == BACKEND_SLEEF) ? sizeof(Sleef_quad) : sizeof(long double);
+
     while (N--) {
-        Sleef_quad in_val = *(Sleef_quad *)in_ptr;
-        T *out_val = (T *)out_ptr;
-        *out_val = from_quad<T>(in_val);
+        quad_value in_val;
+        memcpy(&in_val, in_ptr, elem_size);
+
+        T out_val = from_quad<T>(in_val, backend);
+        memcpy(out_ptr, &out_val, sizeof(T));
```

Comment:
future improvement would be to define both unaligned and aligned loops and only memcpy in the unaligned loops

Reply:
Ahh this must have been missed, looking into that

Reply:
This is resolved